File: word-pattern/word-pattern.py
#leetcode - Word Pattern - easy

class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        s = s.split()
        if len(s) != len(pattern):
            return False
        index_map = {}
        for i in range(len(pattern)):
            if pattern[i] not in index_map:
                index_map[pattern[i]] = s[i]
            else:
                if s[i] != index_map.get(pattern[i]):
                    return False
        return len(set(index_map.values())) == len(index_map.keys())
        

File: word-pattern/word-pattern.js
//leetcode - Word Pattern - easy



/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
     s = s.split(" ");
     if (s.length != pattern.length) {
         return false
     }
     charMap = new Map()
     for (let i = 0; i < pattern.length; i++) {
         if (!charMap.has(pattern[i])) {
             charMap.set(pattern[i], s[i])
         } else {
             if (s[i] != charMap.get(pattern[i])) {
                 return false
             }
         }
     };
     if (charMap.size == new Set(charMap.values()).size) {
         return true
     } else {
         return false
     }
};